python/tvm/contrib/pipeline_executor.py

```diff
@@ -0,0 +1,533 @@
+            if key == "input":
+                return self.input_bindings
+            if key == "output":
+                return self.output_bindings
+
+        raise RuntimeError(f"{key} not found.")
+
+    def get_config(self):
+        """ Get configuration in dictionary format."""
+
+        # Tpological sort to get correct module order in list.
+        self.dag_topology_sort()
+        mconfig = {}
+        for mod in self.mod_wrapper:
+            # Get pipeline configure.
+            mconf = {}
+            output_conf = []
+            module = self.mod_wrapper[mod]
+            for _, binding in module.output_bindings.bindings.items():
+                dep_conf = []
+                output = {}
+                if binding.bindings:
+                    for dep in binding.bindings:
+                        dep_item = {}
+                        _, dname = dep.get_name()
+                        dep_item["mod_idx"] = dep.get_owner_idx()
+                        dep_item["input_name"] = dname
+                        dep_conf.append(dep_item)
+
+                # Ouput_idx start from 0.
+                output["output_idx"] = int(binding.name)
+                output["dependent"] = dep_conf
+                output_conf.append(output)
+
+            mconf["mod_idx"] = module.idx
+            mconf["output"] = output_conf
+
+            # Build module configuration with pipeline and other parameters.
+            mconfig[mod] = {
+                "pipeline": mconf,
+                "target_host": module.target_host,
+                "mod_name": "default",
+                "build": module.build_func,
+                "params": module.params,
+                "target": module.target,
+                "dev": module.dev,
+            }
+
+        return mconfig
+
+    def dag_topology_sort(self):
+        """ Do topological sort to get pipeline module order."""
+        mlist = []
+        mod_wrapper = self.mod_wrapper.copy()
+        while mod_wrapper:
+            temp_list = []
+            for mod, wrapper in mod_wrapper.items():
+                if wrapper.is_root_mod():
+                    temp_list.append(mod)
+                    wrapper.remove_self_from_bindings()
+
+            for mod in temp_list:
+                mod_wrapper.pop(mod, None)
+
+            mlist += temp_list
+
+        for mod, i in zip(mlist, range(len(mlist))):
+            self.mod_wrapper[mod].set_idx_name(i + 1)
+
+    def get_mod_idx(self, mod):
+        """Return idx for specify mod"""
+        idx = self.mod_wrapper[mod].idx
+        return idx
+
+    def pipe_input(self, name):
+        """Return input binding by name"""
+        return self.input_bindings[name]
+
+    def pipe_output(self, idx):
+        """Return output binding by idx"""
+        return self.output_bindings[idx]
+
+
+class PipelineExecutorFactoryModule(object):
+    """This is a factory with list of GraphExecutorFactoryModule and Module configurations.
```

Comment:
What this is called "factory". It is not creating anything. It just stores existing information.

Reply:
fixed.